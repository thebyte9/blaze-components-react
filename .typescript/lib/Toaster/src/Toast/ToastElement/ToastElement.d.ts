/// <reference types="react" />
declare const ToastElement: ({ appearance, placement, transitionDuration, transitionState, ...props }: any) => JSX.Element;
export default ToastElement;
