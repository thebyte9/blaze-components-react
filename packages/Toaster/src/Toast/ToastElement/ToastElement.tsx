import React, { useEffect, useRef, useState } from 'react';
import '@testing-library/jest-dom/extend-expect';

import toastStates from './utils/transitions';

const ToastElement = ({ appearance, placement, transitionDuration, transitionState, ...props }: any) => {
  const [height, setHeight] = useState<number | string>('auto');
  const elementRef: any = useRef(null);

  useEffect(() => {
    if (transitionState === 'entered') {
      const el = elementRef.current;
      setHeight(el.offsetHeight + 8);
    }
    if (transitionState === 'exiting') {
      setHeight(0);
    }
  }, [transitionState]);

  return (
    <div
      data-testid="toast"
      ref={elementRef}
      style={{
        height,
        transition: `height ${transitionDuration - 100}ms 100ms`,
      }}
    >
      <div
        className={`react-toast-notifications__toast react-toast-notifications__toast--${appearance}`}
        style={{
          transition: `transform ${transitionDuration}ms cubic-bezier(0.2, 0, 0, 1), opacity ${transitionDuration}ms`,
          ...toastStates(placement)[transitionState],
        }}
        {...props}
      />
    </div>
  );
};

export default ToastElement;
