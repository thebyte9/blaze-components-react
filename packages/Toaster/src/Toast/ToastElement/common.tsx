const borderRadius = 4;
const gutter = 8;
const toastWidth = 360;

export { borderRadius, gutter, toastWidth };
