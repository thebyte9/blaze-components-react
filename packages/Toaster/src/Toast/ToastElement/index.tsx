import ToastElement from "./ToastElement";

export default ToastElement;
