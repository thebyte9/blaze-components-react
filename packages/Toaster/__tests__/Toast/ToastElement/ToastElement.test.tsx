import { render, screen } from '@testing-library/react';
import expect from 'expect';
import React from 'react';
import ToastElement from '../../../src/Toast/ToastElement';
import '@testing-library/jest-dom/extend-expect';

describe('Toast Element', () => {
  it('should be defined', () => {
    expect(ToastElement).toBeDefined();
  });

  it('should render without throwing error', () => {
    const { asFragment } = render(
      <ToastElement
        appearance="success"
        onMouseEnter={jest.fn()}
        onMouseLeave={jest.fn()}
        placement="top-right"
        transitionDuration={220}
        transitionState="entered"
      >
        <div>children</div>
      </ToastElement>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should set height when transition state is `exiting`', () => {
    render(
      <ToastElement
        appearance="success"
        onMouseEnter={jest.fn()}
        onMouseLeave={jest.fn()}
        placement="top-right"
        transitionDuration={220}
        transitionState="exiting"
      >
        <div>children</div>
      </ToastElement>,
    );

    const toast = screen.getByTestId('toast');
    expect(toast).toHaveStyle('height: 0');
  });
});
