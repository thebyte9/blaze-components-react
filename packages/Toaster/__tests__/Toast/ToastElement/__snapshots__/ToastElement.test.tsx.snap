// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Element should render without throwing error 1`] = `
<DocumentFragment>
  <div
    data-testid="toast"
    style="height: 8px; transition: height 120ms 100ms;"
  >
    <div
      class="react-toast-notifications__toast react-toast-notifications__toast--success"
      style="transition: transform 220ms cubic-bezier(0.2, 0, 0, 1), opacity 220ms; transform: translate3d(0,0,0);"
    >
      <div>
        children
      </div>
    </div>
  </div>
</DocumentFragment>
`;
